AK/NumberFormat.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2020-2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "AK/Assertions.h"
+#include "AK/DeprecatedString.h"
+#include <AK/NumberFormat.h>
+#include <AK/StringView.h>
+
+namespace AK {
+
+// FIXME: Remove this hackery once printf() supports floats.
+static DeprecatedString number_string_with_one_decimal(u64 number, u64 unit, StringView suffix)
+{
+    int decimal = (number % unit) * 10 / unit;
+    return DeprecatedString::formatted("{}.{} {}", number / unit, decimal, suffix);
+}
+
+template<size_t Size>
+static DeprecatedString human_readable_quantity_impl(u64 quantity, const AK::Array<AK::StringView, Size>& suffixes)
```

Comment:
`1024` seems to be used several times in this function as a magic number. Can this be a named constant?

Reply:
It is parameterized now (can be 1024 on 1000)